### PR TITLE
Added Chat System Functionality and corrected navigation tool

### PR DESCRIPTION
**Pull request by LilCSharp**
This is the minimum viable product for week 3. I am currently working on beautifying the comment page.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -14,19 +14,84 @@
...
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+  public void doGet(HttpServletRequest request, HttpServletResponse response) 
+    throws IOException { 
+
+    Query query = new Query("Task").addSort("date", 
+    SortDirection.DESCENDING);
```
Comment (wu-haotian) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: Does this fit in 1 line? Either put in 1 line or add indent on 2nd line.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -14,19 +14,84 @@
...
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content. TODO: modify this file to handle
+comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> list = new ArrayList<String>();
```
Comment (wu-haotian) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Is this list unused?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -14,19 +14,84 @@
...
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(tasks));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) 
+    throws IOException {
+    
+    String text = request.getParameter("chatbox");
```
Comment (wu-haotian) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: is chatbox/namebox 1 word or 2? 

i.e. Should we change chatbox -> chatBox (or nameBox -> namebox)?

**Review by wu-haotian — COMMENTED** [review]
As Antriksh mentioned in today's standup, can you attach a screenshot of the website? I'm not very familiar with web development and this would make my life much easier. Thanks :D 

You can either attach an image attachment, or use go/sniplt to easily attach a link like this: https://screenshot.googleplex.com/oHWCzrwQO6S (but in this way images are only accessible to @google.com users)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (LilCSharp) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I will make those changes.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (LilCSharp) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Yes. That is a remnant from a previous part of the project. I will remove it.

File: portfolio/pom.xml [hunk 4]
```diff
@@ -35,7 +47,7 @@
         <version>2.2.0</version>
         <configuration>
           <!-- TODO: set project ID. -->
```
Comment (antrromet) [inline portfolio/pom.xml]: nit: Remove this TODO.

File: portfolio/src/main/java/com/google/sps/data/Task.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  private final String name;
+  private final String date;
+  private final String text;
+
+  public Task(Long num, String username, String textDate, String words) {
+    id = num;
+    name = username;
+    date = textDate;
+    text = words;
+  }
```
Comment (antrromet) [inline portfolio/src/main/java/com/google/sps/data/Task.java]: These variable names are confusing.

Lets just use id, username, date, msg (description/desc)?

In java the parameters name can be the same as the variable names.
So you'd assign them something like

this.id = id;
this.username = username;

Please update it below as well, where you call this Task.

File: portfolio/src/main/webapp/pages/chat.html [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        
+        <!-- Contact form JS-->
+        <!-- Add in case contacts need be added
+        <script src="assets/mail/jqBootstrapValidation.js"></script>
+        <script src="assets/mail/contact_me.js"></script>
+        -->
+        <!-- Core theme JS-->
+
+        <script src="../js/scripts.js"></script>
+        <script src="../js/chatSystem.js"></script>
```
Comment (antrromet) [inline portfolio/src/main/webapp/pages/chat.html]: Is this a requirement for these scripts to be in the body?

Can't they be in the Head?
Also are all of them needed?

File: portfolio/pom.xml (hunk 4 above)
Comment (LilCSharp) [reply portfolio/pom.xml]: Resolved in following commit

File: portfolio/src/main/java/com/google/sps/data/Task.java (hunk 5 above)
Comment (LilCSharp) [reply portfolio/src/main/java/com/google/sps/data/Task.java]: Resolved in following commit

File: portfolio/src/main/webapp/pages/chat.html (hunk 6 above)
Comment (LilCSharp) [reply portfolio/src/main/webapp/pages/chat.html]: One of them was moved to the head. The other one cannot be moved due to functionality reasons

**Comment by LilCSharp** [comment]
That was removed with the addition of the username feature, as the name quality became non-existent.

**Review by antrromet — APPROVED** [review]

**Pull request merged**